### ISSUE TITLE
Question : More layer of Neural network, will have longer training time and more detail information?? 

### ISSUE THREAD

**Issue by ruah1984** [deepfakes/faceswap#249]
Some question for the neural network. if i trying to have more layer of NN, the training time usually will be longer, and how about the detail of output ? Will it be better than less layer NN??  i not sure if this the theory of NN

**Comment by Kirin-kun** [comment]
I have a hunch that while the compute time will be longer for each epoch, it will get to the result faster if you have more layers & nodes.  If you have a lot of resources, it may be the way to go.

But eventually, the result will be the same, not "more detailed".

**Comment by Clorr** [comment]
More layers is not a guarantee of anything. First you have to know what layer for what purpose. Also not only the layers are important but also the "loss function", the function that says to your model if it behaves right or not.

About layers, the model in DF is an autoencoder with an encoder part and a decoder part.

If you add a layer in encoder you may get more possible variations (but you don't know which ones and it is not guaranteed that you will have useful ones).

If you add layers in decoder, you will have higher resolution output (like what I did in #247 ) but not necessarly more fine grained output. 

If you want more precise output, the way to go is to enhance the loss function. It is the purpose of the GAN and of most of shaoanlu's attempts. In the original model, the loss function is just a mean of differences between generated and expected result, that's what gives this blurry look because the mean does erase details.
